pydexarm/pydexarm.py

```diff
@@ -2,128 +2,258 @@
+        """
         cmd = "M204"+"P" + str(acceleration) + "T"+str(travel_acceleration) + "T" + str(retract_acceleration) + "\r\n"
         self._send_cmd(cmd)
 
-    def set_module_kind(self, kind):
-        self._send_cmd("M888 P" + str(kind) + "\r")
+    def set_module_type(self, module_type):
+        """
+        Set the type of end effector.
+
+        Args:
+            type (str):
+                P0 for Pen holder module
+                P1 Laser engraving module
+                P2 for Pneumatic module
+                P3 for 3D printing module
+        """
+        self._send_cmd("M888 P" + str(module_type) + "\r")
+
+    def get_module_type(self):
+        """
+        Get the type of end effector.
 
-    def get_module_kind(self):
+        Returns:
+            type of the module
+        """
         self.ser.write('M888\r'.encode())
         while True:
-            str = self.ser.readline().decode("utf-8")
-            if len(str) > 0:
-                if str.find("PEN") > -1:
-                    module_kind = 'PEN'
-                if str.find("LASER") > -1:
-                    module_kind = 'LASER'
-                if str.find("PUMP") > -1:
-                    module_kind = 'PUMP'
-                if str.find("3D") > -1:
-                    module_kind = '3D'
-            if len(str) > 0:
-                if str.find("ok") > -1:
-                    return module_kind
-
-    def move_to(self, x, y, z, feedrate=2000):
-        cmd = "G1"+"F" + str(feedrate) + "X"+str(x) + "Y" + str(y) + "Z" + str(z) + "\r\n"
-        self._send_cmd(cmd)
+            serial_str = self.ser.readline().decode("utf-8")
+            if len(serial_str) > 0:
+                if serial_str.find("PEN") > -1:
+                    module_type = 'PEN'
+                if serial_str.find("LASER") > -1:
+                    module_type = 'LASER'
+                if serial_str.find("PUMP") > -1:
+                    module_type = 'PUMP'
+                if serial_str.find("3D") > -1:
+                    module_type = '3D'
+            if len(serial_str) > 0:
+                if serial_str.find("ok") > -1:
+                    return module_type
 
-    def fast_move_to(self, x, y, z, feedrate=2000):
-        cmd = "G0"+"F" + str(feedrate) + "X"+str(x) + "Y" + str(y) + "Z" + str(z) + "\r\n"
-        self._send_cmd(cmd)
+    def move_to(self, x=None, y=None, z=None, feedrate=2000, mode="G1", wait=True):
+        """
+        Move to a cartesian position.
+        Add a linear move to the queue to be performed after all previous moves are completed.
+
+        Args:
+            mode (string, G0 or G1): G1 by default. use G0 for fast mode
+            x, y, z (int): The position, in millimeters by default. Units may be set to inches by G20. Note that the center of y axis is 300mm.
+            feedrate (int): set the feedrate for all subsequent moves
+        """
+        cmd = mode + "F" + str(feedrate)
+        if x:
```

Comment:
by default x,y,z is None. This way allows to set only one or two of the aixs like:
G1F2000Y320